contrib/intel/jenkins/tests.py

```diff
@@ -1073,27 +1073,40 @@ def __init__(self, jobname, buildno, testname, hw, core_prov, fabric,
             'MLX5_SCATTER_TO_CQE'     : '0'
         }
 
-        self.tests = {
-                'H2H'   : [
-                            'write',
-                            'read',
-                            'send'
-                        ],
-                'H2D'   : [
-                            'write',
-                            'read',
-                            'send'
-                        ],
-                'D2H'   : [
-                            'write',
-                            'read',
-                            'send'
-                        ],
-                'D2D'   : [
-                            'write',
-                            'read',
-                            'send'
-                        ]
+        self.single_node_combinations = {
+                'H2H'   : {
+                            'H' : 'H'
+                        },
+                'H2D'   : {
+                            'H' : ['-d 0', '-d 1']
+                        },
+                'D2H'   : {
+                            '-d 0' : 'H',
+                            '-d 1' : 'H'
+                        },
+                'D2D'   : {
+                            '-d 0' : ['-d 1', '-d 2', '-d 3'],
+                            '-d 1' : ['-d 2', '-d 3']
+                        }
```

Comment:
The control logic later would be much simplified if the test configuration are defined as command line options directly and the lists are expanded. e.g.:

```
      'H2D'  : {
              '-m malloc' : '-m device -d 0'
              '-m malloc' : '-m device -d 1'
       },
```

Reply:
I figured that adding -m device and -m malloc to each config would be redundant and hence just specified H for malloc  and -d <device_no> for device. This way any combination could be added later on in the config.

Reply:
That results in the triple loop control structure that is much harder to understand vs the single loop if the redundancy is tolerated in the configuration. IMHO, the configuration with the `-m` option added is also easier to read.